build.rs

```diff
@@ -27,4 +27,36 @@ fn main() {
     }
     let target = env::var("TARGET").unwrap();
     println!("cargo:rustc-env=TARGET={target}");
+
+    let target_os = env::var("CARGO_CFG_TARGET_OS");
+    let target_env = env::var("CARGO_CFG_TARGET_ENV");
+    if Ok("windows") == target_os.as_deref() && Ok("msvc") == target_env.as_deref() {
```

Comment:
Nit: please switch the constants with the run-time values, which seems more idiomatic. I'd also like to the invert the condition here to do an early return for non-Windows-MSVC targets.

Next, please add some comments here similar to the PR description to describe what problem we're trying to solve here.

Reply:
Do you mean something like:

```rust
if target_os.as_deref() != Ok("windows") && target_env.as_deref() != Ok("msvc") {
    return;
}
```

Reply:
Yup!